azurerm/resource_arm_private_link_service.go

```diff
@@ -0,0 +1,649 @@
+			return fmt.Errorf("Error setting `auto_approval_subscription_ids`: %+v", err)
+		}
+		if err := d.Set("visibility_subscription_ids", flattenArmPrivateLinkServicePrivateLinkServicePropertiesVisibility(privateLinkServiceProperties.Visibility)); err != nil {
+			return fmt.Errorf("Error setting `visibility_subscription_ids`: %+v", err)
+		}
+		if err := d.Set("fqdns", utils.FlattenStringSlice(privateLinkServiceProperties.Fqdns)); err != nil {
+			return fmt.Errorf("Error setting `fqdns`: %+v", err)
+		}
+		if err := d.Set("nat_ip_configuration", flattenArmPrivateLinkServicePrivateLinkServiceIPConfiguration(privateLinkServiceProperties.IPConfigurations)); err != nil {
+			return fmt.Errorf("Error setting `nat_ip_configuration`: %+v", err)
+		}
+		if err := d.Set("load_balancer_frontend_ip_configuration_ids", flattenArmPrivateLinkServiceFrontendIPConfiguration(privateLinkServiceProperties.LoadBalancerFrontendIPConfigurations)); err != nil {
+			return fmt.Errorf("Error setting `load_balancer_frontend_ip_configuration_ids`: %+v", err)
+		}
+		if err := d.Set("network_interface_ids", flattenArmPrivateLinkServiceInterface(privateLinkServiceProperties.NetworkInterfaces)); err != nil {
+			return fmt.Errorf("Error setting `network_interface_ids`: %+v", err)
+		}
+		if err := d.Set("private_endpoint_connection", flattenArmPrivateLinkServicePrivateEndpointConnection(privateLinkServiceProperties.PrivateEndpointConnections)); err != nil {
+			return fmt.Errorf("Error setting `private_endpoint_connection`: %+v", err)
+		}
+	}
+	d.Set("type", resp.Type)
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmPrivateLinkServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).network.PrivateLinkServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["privateLinkServices"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmPrivateLinkServicePrivateLinkServicePropertiesAutoApproval(input []interface{}) *network.PrivateLinkServicePropertiesAutoApproval {
+	if len(input) == 0 {
+		return nil
+	}
+
+	subscriptions := make([]string, 0)
+
+	for _, v := range input {
+		subscriptions = append(subscriptions, v.(string))
+	}
+
+	result := network.PrivateLinkServicePropertiesAutoApproval{
+		Subscriptions: &subscriptions,
```

Comment:
We have a utils function `FlattenStringSlice` that will make the slice for us, that should allow us to pull this out?

Reply:
Fixed, but I kept the function as explained above.